form.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+/**
+ * Sets the change handler for the cluster select.
+ */
+function set_cluster_change_handler() {
+
+  // Ensure owens does not use spark 2.4.5 on page load.
+  toggle_options("batch_connect_session_context_spark_version");
+  // Ensure pitzer does not use jupyter_app 3.0 on page load.
+  toggle_options("batch_connect_session_context_version");
+
+  const cluster_input = $('#batch_connect_session_context_cluster');
+
+  cluster_input.change((event) => {
+    // spark version change.
+    toggle_options("batch_connect_session_context_spark_version");
+    // jupyter_app version change.
+    toggle_options("batch_connect_session_context_version");
+  });
+}
+
+function toggle_options(element_name) {
+  const cluster = current_cluster_capitalized();
+  const search = "#" + element_name + " option";
+  const options = $(search);
+
+  options.each(function(_i, option) {
+    // the variable 'option' is just a data structure. it has no attr, data, show
+    // or hide methods so we have to query for it again
+    let option_element = $(search + "[value='" + option.value + "']");
+    let data = option_element.data();
+    let show = data["optionFor" + cluster];
+
+    if (cluster === "Owens") {
+      let pitzer_only_spark_version = '2.4.5';
+      option_element.prop('disabled', false);
+      // disabling spark version not on owens.
+      if (option.value == pitzer_only_spark_version) {
```

Comment:
We shouldn't have clusternames or versions here I don't believe. 


```javascript
let show = data["optionFor" + cluster];
```
This bit here should be `dataOptionForOwens` attached to version 2.4.5. 

What I mean is, you let the data attributes determine the logic, not extra logic to checking for === "Owens" and so on.

Reply:
Done, and I also wrote it to not include all the true values in the form.yml.erb so that only the false values need to be kept track of like you had pointed out.